doc/source/whatsnew/v3.0.0.rst

```diff
@@ -546,6 +546,7 @@ Interval
 Indexing
 ^^^^^^^^
 - Bug in :meth:`DataFrame.__getitem__` returning modified columns when called with ``slice`` in Python 3.12 (:issue:`57500`)
+- Bug in :meth:`DataFrame.from_records()` throwing a ValueError when passed an empty list as an index (:issue:`58594`)
```

Comment:
```suggestion
- Bug in :meth:`DataFrame.from_records` throwing a ``ValueError`` when passed an empty list in ``index`` (:issue:`58594`)
```

Reply:
@mroeschke, I just pushed a commit to address all your feedback. Mind taking another look?

Thanks!